app/models/miq_ae_instance.rb

```diff
@@ -149,8 +149,9 @@ def self.display_name(number = 1)
   end
 
   def self.find_best_match_by(user, relative_path)
-    domain_ids = user.current_tenant.enabled_domains.collect(&:id)
-    joins(:domain).where(:miq_ae_namespaces => {:id => domain_ids}).order("miq_ae_namespaces.priority DESC")
+    domain_ids = user.current_tenant.enabled_domains.select(&:id)
```

Comment:
Can this be pluck instead of select?

Reply:
@Fryguy This is a subquery. `.select` would make this one query instead of two.

Reply:
before we spend too much time on this, please note that this week we're going to use local ae domain id -> domain name lookups. We added 400+ domain queries to provisioning workflow.

Reply:
> This is a subquery. 

Is it?  .select with a block is running Enumerable#select (not ActiveRecord::Relation#select) and is essentially saying "get me all of the domains that have an id value" (which will be all of them)

Reply:
If you want to use a subquery I think you can just remove the `.select(&:id)` entirely

Reply:
oooh - that is totally running it locally. good eye